ext/compressors/zlib/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+# See the file LICENSE for redistribution information.
+#
+
+cmake_minimum_required(VERSION 3.11.0)
+
+project(zlib C)
+
+config_bool(
+    HAVE_BUILTIN_EXTENSION_ZLIB
+    "Builtin zlib compression library"
+    DEFAULT OFF
+    DEPENDS "HAVE_LIBZ"
+    DEPENDS_ERROR ON "Failed to find zlib library"
+)
+
+if (HAVE_BUILTIN_EXTENSION_ZLIB AND ENABLE_LIBZ)
+    message(FATAL_ERROR "Only one of 'ENABLE_LIBZ' or 'HAVE_BUILTIN_EXTENSION_ZLIB' can be enabled")
+endif()
+
+set(sources "zlib_compress.c")
+set(link_type)
+if(HAVE_BUILTIN_EXTENSION_LZ4)
```

Comment:
These checks should be based on ZLIB?

Reply:
Oh amazing catch! That's correct, will fix.
(I may have been to comfortable with copy/paste when writing the `ext/compressors` CMakeLists)

Reply:
Fixed!